runtime/gateway.go

```diff
@@ -333,10 +333,14 @@ func (gateway *Gateway) setupMetrics(config *StaticConfig) (err error) {
 	)
 	// As per M3 best practices, creating separate all-host and per-host metrics
 	// to reduce metric cardinality when querying metrics for all hosts.
-	gateway.AllHostScope = gateway.RootScope
-	gateway.PerHostScope = gateway.RootScope.Tagged(
+	gateway.AllHostScope = gateway.RootScope.SubScope(
+		service + "." + env + ".all-workers",
```

Comment:
You can add the service Prefix to tally.ScopeOptions instead when creating the RootScope.
```
	gateway.RootScope, gateway.scopeCloser = tally.NewRootScope(
		tally.ScopeOptions{
			Prefix:         service,
			Tags:           defaultTags,
			CachedReporter: gateway.metricsBackend,
			Separator:      tally.DefaultSeparator,
		},
		time.Duration(config.MustGetInt("metrics.flushInterval"))*time.Millisecond,
	)
```

Reply:
we still have to create two sub scopes. its the same thing. The explicit sub scope call means the prefix is in a single place instead of concatenated from two different places.

Reply:
Sounds good.